### PR TITLE
refactor: `RelationalBone`s `refKeys` and `parentKeys` as set

### PR DESCRIPTION
**Pull request by phorward**
Tested. Might collide with #1022.

### CODE REVIEW

**Review by ArneGudermann — APPROVED** [review]

File: src/viur/core/bones/relational.py [hunk 1]
```diff
@@ -600,7 +592,7 @@ def postSavedHandler(self, skel, boneName, key):
                 dbObj["viur_delayed_update_tag"] = time()
                 dbObj["viur_relational_updateLevel"] = self.updateLevel.value
                 dbObj["viur_relational_consistency"] = self.consistency.value
-                dbObj["viur_foreign_keys"] = self.refKeys
+                dbObj["viur_foreign_keys"] = tuple(self.refKeys)
```
Comment (phorward) [inline src/viur/core/bones/relational.py:595]: Due https://github.com/viur-framework/viur-datastore/issues/46, we need to use `list()` here.
```suggestion
                dbObj["viur_foreign_keys"] = list(self.refKeys)
```

File: src/viur/core/bones/relational.py [hunk 2]
```diff
@@ -619,7 +611,7 @@ def postSavedHandler(self, skel, boneName, key):
             dbObj["viur_dest_kind"] = self.kind
             dbObj["viur_relational_updateLevel"] = self.updateLevel.value
             dbObj["viur_relational_consistency"] = self.consistency.value
-            dbObj["viur_foreign_keys"] = self.refKeys
+            dbObj["viur_foreign_keys"] = tuple(self.refKeys)
```
Comment (phorward) [inline src/viur/core/bones/relational.py:614]: Due https://github.com/viur-framework/viur-datastore/issues/46, we need to use `list()` here.
```suggestion
                dbObj["viur_foreign_keys"] = list(self.refKeys)
```

File: src/viur/core/bones/relational.py [hunk 3]
```diff
@@ -265,17 +255,19 @@ def __init__(
         if self.kind is None or self.module is None:
             raise NotImplementedError("Type and Module of RelationalBone must not be None")
 
+        # Referenced keys
+        self.refKeys = {"key"}
```
Comment (sveneberth) [inline src/viur/core/bones/relational.py:259]:
```suggestion
        self.refKeys = {"key", "name"}
```

File: src/viur/core/bones/relational.py [hunk 4]
```diff
@@ -265,17 +255,19 @@ def __init__(
...
         if refKeys:
-            if not "key" in refKeys:
-                refKeys.append("key")
-            self.refKeys = refKeys
+            self.refKeys |= set(refKeys)
 
+        # Parent keys
+        self.parentKeys = {"key"}
```
Comment (sveneberth) [inline src/viur/core/bones/relational.py:264]:
```suggestion
        self.parentKeys = {"key", "name"}
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

File: src/viur/core/bones/relational.py (hunk 3 above)
Comment (phorward) [reply src/viur/core/bones/relational.py:259]: No, see the default refKeys definition.
There might also be cases, where a `name` is not available.

**Review by sveneberth — APPROVED** [review]

**Pull request merged**